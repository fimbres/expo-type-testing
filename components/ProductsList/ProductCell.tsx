import React, { FC } from 'react'
import { StyleSheet, Text, View, Image, Pressable } from 'react-native'
import { Entypo } from '@expo/vector-icons';
import Numeral from "numeral";
import moment from 'moment';
import 'moment/locale/es';

import { IProduct } from '../../types'
import { Colors, Styles } from '../../constants';

interface ProductCellProps {
    product: IProduct;
    onPress: () => void;
}

export const ProductCell: FC<ProductCellProps> = ({ product, onPress }) => {
    const { image, product: name, createdAt, points, is_redemption } = product;

    const getDateLabel = () => {
        const momentDate = moment(createdAt);
        return `${momentDate.format("d")} de ${momentDate.format("MMMM, YYYY")}`
    }

    return (
        <Pressable style={styles.cell} onPress={onPress}>
            <View style={styles.container}>
                <Image source={{ uri: image }} style={styles.image}/>
                <View>
                    <Text style={styles.primaryText}>{name}</Text>
                    <Text style={Styles.textCaptionExtraSmall}>{getDateLabel()}</Text>
                </View>
            </View>
            <View style={styles.container}>
                <Text style={styles.pointsText}>
                    <Text style={{ color: is_redemption ? Colors.red : Colors.green }}>{is_redemption ? "-" : "+"}</Text>
                    {Numeral(points).format("0,0")}
                </Text>
                <Entypo name="triangle-right" size={14} color="black" />
            </View>
        </Pressable>
    )
}

const styles = StyleSheet.create({
    cell: {
        width: '100%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginBottom: 8,
        paddingHorizontal: 10,
    },
    container: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    image: {
        width: 55,
        height: 55,
        borderRadius: 10,
        marginRight: 11,
    },
    primaryText: {
        ...Styles.textTitleSmall,
        marginBottom: 7,
    },
    pointsText: {
        ...Styles.textTitleRegular,
        marginRight: 10
    }
})