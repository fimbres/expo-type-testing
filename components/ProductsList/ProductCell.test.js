import React from 'react';
import renderer from 'react-test-renderer';

import { ProductCell } from './ProductCell';

const testProduct = {
    createdAt: "2022-12-08T18:54:56.243Z",
    id: 5,
    image: "https://loremflickr.com/640/480/people",
    is_redemption: true,
    points: 69814,
    product: "Rustic Rubber Bacon",
};

describe('<ProductCell />', () => {
    it('has 2 child', () => {
        const tree = renderer
            .create(
                <ProductCell
                    product={testProduct}
                    onPress={() => console.log("onPress")}
                />
            ).toJSON();
        expect(tree.children.length).toBe(2);
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <ProductCell
                    product={testProduct}
                    onPress={() => console.log("onPress")}
                />
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
