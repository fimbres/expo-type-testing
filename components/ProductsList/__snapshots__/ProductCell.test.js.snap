// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCell /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 8,
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://loremflickr.com/640/480/people",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 55,
          "marginRight": 11,
          "width": 55,
        }
      }
    />
    <View>
      <Text
        style={
          Object {
            "color": "#020202",
            "fontFamily": "avenir-black",
            "fontSize": 14,
            "fontWeight": "800",
            "lineHeight": 19.12,
            "marginBottom": 7,
          }
        }
      >
        Rustic Rubber Bacon
      </Text>
      <Text
        style={
          Object {
            "color": "#020202",
            "fontFamily": "avenir-book",
            "fontSize": 12,
            "fontWeight": "400",
            "lineHeight": 16.39,
          }
        }
      >
        4 de diciembre, 2022
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#020202",
          "fontFamily": "avenir-black",
          "fontSize": 16,
          "fontWeight": "800",
          "lineHeight": 21.86,
          "marginRight": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
          }
        }
      >
        -
      </Text>
      69,814
    </Text>
    <Text />
  </View>
</View>
`;
